--- squeezelite-1.8/pcm.c.orig	2016-03-07 12:51:58.216427031 +0000
+++ squeezelite-1.8/pcm.c	2016-03-07 12:53:23.690094348 +0000
@@ -204,7 +204,6 @@
 		LOCK_O_not_direct;
 		output.track_start = outputbuf->writep;
 		decode.new_stream = false;
-		output.next_sample_rate = decode_newstream(sample_rate, output.supported_rates);
 #if DSD
 		if (sample_size == 3 &&
 			is_stream_dop(((u8_t *)streambuf->readp) + (bigendian?0:2),
@@ -215,12 +214,15 @@
 				output.next_fmt = output.dsdfmt;
 			else
 				output.next_fmt = DOP;
+			output.next_sample_rate = sample_rate;
 			output.fade = FADE_INACTIVE;
 		} else {
+			output.next_sample_rate = decode_newstream(sample_rate, output.supported_rates);
 			output.next_fmt = PCM;
 			if (output.fade_mode) _checkfade(true);
 		}
 #else
+		output.next_sample_rate = decode_newstream(sample_rate, output.supported_rates);
 		if (output.fade_mode) _checkfade(true);
 #endif
 		UNLOCK_O_not_direct;
