diff -ruN squeezelite-1.8.orig/main.c squeezelite-1.8/main.c
--- squeezelite-1.8.orig/main.c	2016-11-17 13:27:23.395952990 +0000
+++ squeezelite-1.8/main.c	2016-11-17 13:27:01.180069877 +0000
@@ -203,6 +203,7 @@
 	char *name = NULL;
 	char *namefile = NULL;
 	char *modelname = NULL;
+	char *firmware = NULL;
 	char *logfile = NULL;
 	u8_t mac[6];
 	unsigned stream_buf_size = STREAMBUF_SIZE;
@@ -365,7 +366,8 @@
 			}
 			break;
 		case 'M':
-			modelname = optarg;
+			modelname = next_param(optarg, ':');
+			firmware = next_param(NULL, ':');
 			break;
 		case 'r':
 			{ 
@@ -633,7 +635,7 @@
 		exit(1);
 	}
 
-	slimproto(log_slimproto, server, mac, name, namefile, modelname, lms_downsample);
+	slimproto(log_slimproto, server, mac, name, namefile, modelname, firmware, lms_downsample);
 
 	decode_close();
 	stream_close();
diff -ruN squeezelite-1.8.orig/slimproto.c squeezelite-1.8/slimproto.c
--- squeezelite-1.8.orig/slimproto.c	2016-11-17 13:27:23.395952990 +0000
+++ squeezelite-1.8/slimproto.c	2016-11-17 13:28:11.860910378 +0000
@@ -111,7 +111,7 @@
 }
 
 static void sendHELO(bool reconnect, const char *fixed_cap, const char *var_cap, u8_t mac[6]) {
-	const char *base_cap = "Model=squeezelite,AccuratePlayPoints=1,HasDigitalOut=1,HasPolarityInversion=1,Firmware=" VERSION;
+	const char *base_cap = "Model=daphile,AccuratePlayPoints=1,HasDigitalOut=1,HasPolarityInversion=1,Firmware=" VERSION;
 	struct HELO_packet pkt;
 
 	memset(&pkt, 0, sizeof(pkt));
@@ -855,10 +855,10 @@
 	return;
 }
 
-#define FIXED_CAP_LEN 256
+#define FIXED_CAP_LEN 320
 #define VAR_CAP_LEN   128
 
-void slimproto(log_level level, char *server, u8_t mac[6], const char *name, const char *namefile, const char *modelname, bool lms_downsample) {
+void slimproto(log_level level, char *server, u8_t mac[6], const char *name, const char *namefile, const char *modelname, const char *firmware, bool lms_downsample) {
 	struct sockaddr_in serv_addr;
 	static char fixed_cap[FIXED_CAP_LEN], var_cap[VAR_CAP_LEN] = "";
 	bool reconnect = false;
@@ -911,7 +911,7 @@
 	if (!running) return;
 
 	LOCK_O;
-	snprintf(fixed_cap, FIXED_CAP_LEN, ",ModelName=%s,MaxSampleRate=%u", modelname ? modelname : MODEL_NAME_STRING,
+	snprintf(fixed_cap, FIXED_CAP_LEN, ",Firmware=%s,ModelName=%s,MaxSampleRate=%u", firmware ? firmware : VERSION, modelname ? modelname : MODEL_NAME_STRING,
 			 lms_downsample ? output.supported_rates[0] : ref[0]);
 	
 	for (i = 0; i < MAX_CODECS; i++) {
diff -ruN squeezelite-1.8.orig/squeezelite.h squeezelite-1.8/squeezelite.h
--- squeezelite-1.8.orig/squeezelite.h	2016-11-17 13:27:23.395952990 +0000
+++ squeezelite-1.8/squeezelite.h	2016-11-17 13:27:01.180069877 +0000
@@ -23,7 +23,7 @@
 #define VERSION "v1.8+ (Daphile)"
 
 #if !defined(MODEL_NAME)
-#define MODEL_NAME SqueezeLite
+#define MODEL_NAME Daphile
 #endif
 
 #define QUOTE(name) #name
@@ -426,7 +426,7 @@
 void buf_destroy(struct buffer *buf);
 
 // slimproto.c
-void slimproto(log_level level, char *server, u8_t mac[6], const char *name, const char *namefile, const char *modelname, bool lms_downsample);
+void slimproto(log_level level, char *server, u8_t mac[6], const char *name, const char *namefile, const char *modelname, const char *firmware, bool lms_downsample);
 void slimproto_stop(void);
 void wake_controller(void);
 void send_jsonrpc(char *);
